torch/_inductor/fx_utils.py

```diff
@@ -87,7 +87,9 @@ def is_fake_tensor_same(new, old):
                 return all(
                     is_fake_tensor_same(new_i, old_i) for new_i, old_i in zip(new, old)
                 )
-            assert isinstance(new, torch.Tensor)
+            if not isinstance(new, torch.Tensor):
+                assert isinstance(new, (torch.SymInt, torch.SymBool, torch.SymFloat))
+                return new == old
```

Comment:
This will generate a guard, what we probably want is `statically_known_equals(new.node.expr, old.node.expr)`.

Reply:
Although that said, `new.stride() != old.stride()` below this probably generates guards as well...

Reply:
`statically_known_equals` makes sense in this context IMO